lib/pages/edit_profile_page.dart

```diff
@@ -40,14 +45,44 @@ class EditProfileForm extends StatefulWidget {
   EditProfileFormState createState() => EditProfileFormState();
 }
 
-class EditProfileFormState extends State<EditProfileForm> {
+class  EditProfileFormState  extends State<EditProfileForm> {
   final _formKey = GlobalKey<FormState>();
 
+  String _username = "";
+  String _firstName = "";
+  String _lastName = "";
+  String _bio = "";
   DateTime _birthday = DateTime.now();
   bool _isPrivate = false;
 
+
+ /* Future<void> databaseAccess() async {
+    try {
+      FirebaseFirestore db = FirebaseFirestore.instance;
+      CollectionReference profileRef = db.collection("profile_data");
+      final FirebaseAuth auth = FirebaseAuth.instance;
+      final currentUser = auth.currentUser;
+      if (currentUser != null) {
+        DocumentSnapshot qs = await profileRef.doc(currentUser.uid).get();
+        if (qs.exists) {
+          //access specific data :
+          Map<String, dynamic> d = qs.data() as Map<String, dynamic>;
+          print(d['name']);
+          _username = d['username.profile_name'];
+          print(_username);
+          _firstName = d['name'];
+          _lastName = d['surname'];
+          _bio = d['bio'];
+          _birthday = d['birthday'];
+          _isPrivate = d['visibility'];
+        }
+      }
+    } catch (e) {   }
+  }*/
+
```

Comment:
Do you want to keep this here for the future?

Reply:
yes, originally I wanted to populate the text inputs with the current data on the database , but I ran into quite a few issues , so I decided I will work on this when I get the time , since its not really a priority

Reply:
Okay cool and just to confirm the code is supposed to update info in your profile on FireStore right?

Reply:
yes , does it not do that when you try it?

Reply:
I'm struggling to test it because I don't have a user that has profile_data. Trying to create a new one.

Reply:
Okay I logged in with you account, maybe I'm doing something wrong, will send a video.

Reply:
please do